mon-pix/app/services/session.js

```diff
@@ -83,6 +74,41 @@ export default class CurrentSessionService extends SessionService {
     this.attemptedTransition = transition;
   }
 
+  get isAuthenticatedByGar() {
+    return Boolean(this.externalUserTokenFromGar);
+  }
+
+  get externalUserTokenFromGar() {
+    return this.data?.externalUser;
+  }
+
+  set externalUserTokenFromGar(token) {
+    this.data.externalUser = token;
+  }
+
+  get userIdForAssociation() {
+    return this.data?.expectedUserId;
+  }
+
+  set userIdForAssociation(userId) {
+    this.data.expectedUserId = userId;
+  }
+
+  revokeGarExternalUserToken() {
+    if (this.externalUserTokenFromGar) {
+      delete this.data.externalUser;
+    }
+  }
+
+  revokeGarAuthenticationContext() {
+    if (this.userIdForAssociation) {
+      delete this.data.expectedUserId;
+    }
+    if (this.externalUserTokenFromGar) {
+      delete this.data.externalUser;
+    }
```

Comment:
En terme de cohérence je trouve étrange que dans certains cas il y ait `this.data?.` et dans d'autres `this.data.`.

Reply:
En effet, `data` semble être toujours déclaré, je peux enlever le `?`